File: 121-best-time-to-buy-and-sell-stock/java/Solution.java
class Solution {
    public int maxProfit(int[] prices) {
        int boughtDay = prices[0];
        int profit = 0;
        int current_profit = 0;
        for(int i = 0; i < prices.length ; i++)
        {
            if(prices[i] < boughtDay)
            {
                boughtDay = prices[i];
            }
            
            current_profit = prices[i] - boughtDay;
            
            if(profit < current_profit)
            {
                profit = current_profit;
            }
        }
        
        return profit;
    }
}

File: 121-best-time-to-buy-and-sell-stock/php/Solution.php
<?php

class Solution {

    /**
     * @param Integer[] $prices
     * @return Integer
     */

    function maxProfit($prices) {

        $buy = $prices[0];
        $maxProfit = 0;

        for($i = 1 ; $i < count($prices) ; $i++){

            $currBuy = min($buy , $prices[$i]);
            $maxProfit = max($prices[$i] - $currBuy , $maxProfit);

            $buy = $currBuy;
        }

        return $maxProfit;
    }
}